### PR TITLE
[Auth] Refactor IAM auth

### PR DESCRIPTION
**Pull request by aleoli**
# Description

This pr makes it possible to manage IAM users with the new authentication

# How Has This Been Tested?

- [x] on EKS

### CODE REVIEW

File: cmd/liqo-controller-manager/main.go [hunk 1]
```diff
@@ -151,6 +152,12 @@ func main() {
 	foreignClusterPingTimeout := flag.Duration("foreign-cluster-ping-timeout", 5*time.Second,
 		"The timeout of the ForeignCluster API server readiness check")
 
+	// AWS configurations
+	flag.StringVar(&awsConfig.AwsAccessKeyID, "aws-access-key-id", "", "AWS IAM AccessKeyID for the Liqo User")
+	flag.StringVar(&awsConfig.AwsSecretAccessKey, "aws-secret-access-key", "", "AWS IAM SecretAccessKey for the Liqo User")
+	flag.StringVar(&awsConfig.AwsRegion, "aws-region", "", "AWS region where the local cluster is running")
+	flag.StringVar(&awsConfig.AwsClusterName, "aws-cluster-name", "", "Name of the local EKS cluster")
```
Comment (fra98) [inline cmd/liqo-controller-manager/main.go:159]: Can you move these flags few lines below with the other authentication flags?

File: pkg/identityManager/const.go [hunk 2]
```diff
@@ -28,6 +26,8 @@ const (
 	remoteClusterIDTagKey = "liqo.io/remote-cluster-id"
 	managedByTagKey       = "liqo.io/managed-by"
 	managedByTagValue     = "liqo"
+	identityTypeTagKey    = "liqo.io/identity-type"
```
Comment (fra98) [inline pkg/identityManager/const.go:29]: We could re-use the identical label in `pkg/consts/authentication.go`

**Review by fra98 — APPROVED** [review]

File: pkg/identityManager/const.go (hunk 2 above)
Comment (aleoli) [reply pkg/identityManager/const.go]: I copied the value from this constant to the other, in order to continue using the tagkey suffix, is it ok for you?

**Pull request merged**